clar2wasm/src/wasm_generator.rs

```diff
@@ -1,22 +1,46 @@
+use clarity::vm::functions::define::DefineFunctions;
+use clarity::vm::types::{QualifiedContractIdentifier, TraitIdentifier};
+use clarity::vm::ClarityVersion;
 use clarity::vm::{
     analysis::ContractAnalysis,
     diagnostic::DiagnosableError,
     functions::NativeFunctions,
+    representations::{Span, TraitDefinition},
     types::{
         CharType, FunctionType, PrincipalData, SequenceData, SequenceSubtype, StringSubtype,
         TypeSignature,
     },
     variables::NativeVariables,
-    ClarityName, SymbolicExpression, SymbolicExpressionType,
+    ClarityName, SymbolicExpression, SymbolicExpressionType, Value,
 };
+use lazy_static::lazy_static;
 use std::{borrow::BorrowMut, collections::HashMap};
 use walrus::{
     ir::{BinaryOp, Block, InstrSeqType, LoadKind, MemArg, StoreKind, UnaryOp},
     ActiveData, DataKind, FunctionBuilder, FunctionId, GlobalId, InstrSeqBuilder, LocalId, Module,
     ValType,
 };
 
-use crate::ast_visitor::{traverse, ASTVisitor};
+pub type CResult<T> = Result<T, Box<dyn std::error::Error>>;
+
+#[derive(Debug)]
+struct GenericError;
```

Comment:
Did you try to use the `GeneratorError` instead of adding this new error type? I think that `error` field could probably be removed from the struct and used in place of this one.